jsonschema-module-jackson/src/main/java/com/github/victools/jsonschema/module/jackson/JacksonModule.java

```diff
@@ -248,4 +252,19 @@ protected boolean shouldIgnoreField(FieldScope field) {
         return beanDescription.findProperties().stream()
                 .noneMatch(propertyDefinition -> fieldName.equals(propertyDefinition.getInternalName()));
     }
+
+    /**
+     * Look-up the given field's {@link JsonProperty} annotation and its "required" attribute,  and calculate wether or not this field should be part of the required properties
+     *
+     * @param field field to look-up required strategy for
+     * @return whether the field should be in the "required" list or not
+     */
+    private boolean getRequiredCheckBasedOnJsonPropertyAnnotation(FieldScope field) {
+        JsonProperty jsonProperty = field.getAnnotationConsideringFieldAndGetter(JsonProperty.class) ;
+        if ( jsonProperty == null )
+            return false;
+        else
+            return jsonProperty.required();
```

Comment:
checkstyle complains about missing curly braces. But we can shorten this a bit by getting rid of the `if`-`else` blocks:
```suggestion
        return jsonProperty != null && jsonProperty.required();
```